src/main/java/edu/kit/hci/soli/config/template/JteContext.java

```diff
@@ -85,4 +88,21 @@ public String format(LocalTime time) {
     public PageSpec page(@PropertyKey(resourceBundle = "messages") String key) {
         return new PageSpec(lookup(key), "SOLI");
     }
+
+    public String getUserName(LoginStateModel login) {
+        switch (login.kind()) {
+            case VISITOR -> {return this.lookup("user.visitor");}
+            case GUEST -> {return this.lookup("user.guest");}
+            case ADMIN -> {return this.lookup("user.admin");}
+            default -> {return login.name();}
+        }
+    }
+
+    public String getUserName(User user) {
+        if (Objects.equals(user.getUsername(), "Guest")) {
```

Comment:
This should be based on the implementation of this check in `UserService`, not an ad-hoc check using the username. The service can be injected indirectly through the constructor.
Also, admin users should be handled.